### PR TITLE
[Object_Detection] Add MultiScaleAnchorGenerator and docstring for An…

### PR DESCRIPTION
**Pull request by tanzhenyu**
[Object_Detection] Add MultiScaleAnchorGenerator

### CODE REVIEW

**Comment by bhack** [comment]
Wow +611 -124 :smile:

File: kerascv/layers/anchor_generators/anchor_generator.py [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 import tensorflow as tf
 
 
 class AnchorGenerator(tf.keras.layers.Layer):
```
Comment (bhack) [inline kerascv/layers/anchor_generators/anchor_generator.py:18]: Was this base class designed over the experience of https://github.com/tensorflow/models/blob/master/research/object_detection/core/anchor_generator.py? Or is this a from scratch design?

File: kerascv/layers/anchor_generators/multi_scale_anchor_generator.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from .anchor_generator import AnchorGenerator
+
+
+class MultiScaleAnchorGenerator(tf.keras.layers.Layer):
```
Comment (bhack) [inline kerascv/layers/anchor_generators/multi_scale_anchor_generator.py:19]: Same here for https://github.com/tensorflow/models/blob/master/research/object_detection/anchor_generators/multiscale_grid_anchor_generator.py#L30

File: kerascv/layers/anchor_generators/anchor_generator.py (hunk 1 above)
Comment (tanzhenyu) [reply kerascv/layers/anchor_generators/anchor_generator.py:18]: Good question!
This is mostly a scratch design for proof of concept at this moment. Francois and I will send out RFC for public comments and feedbacks later :-)

File: kerascv/layers/anchor_generators/multi_scale_anchor_generator.py (hunk 2 above)
Comment (tanzhenyu) [reply kerascv/layers/anchor_generators/multi_scale_anchor_generator.py:19]: Same above

**Comment by tanzhenyu** [comment]
> Wow +611 -124 😄

Just moving files and adding comprehensive tests :-)

File: kerascv/layers/anchor_generators/anchor_generator.py (hunk 1 above)
Comment (bhack) [reply kerascv/layers/anchor_generators/anchor_generator.py:18]: Ok cause there are a few derived classes at https://github.com/tensorflow/models/tree/master/research/object_detection/anchor_generators

File: kerascv/layers/anchor_generators/anchor_generator.py [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+            For example, `anchor_stride=(30, 40)` means each anchor is separated by 30 pixels in height, and
+            40 pixels in width. Defaults to `None`, where anchor stride would be calculated as
+            `min(image_height, image_width) / feature_map_height` and
+            `min(image_height, image_width) / feature_map_width`.
+        anchor_offset: A list/tuple of 2 floats between [0., 1.] representing the center of anchor points relative to
+            the upper-left border of each feature map cell. Defaults to `None`, which is the center of each
+            feature map cell when `anchor_stride=None`, or center of anchor stride otherwise.
+        clip_boxes: Boolean to represents whether the anchor coordinates should be clipped to the image size.
+            Defaults to `True`.
+        norm_coords: Boolean to represents whether the anchor coordinates should be normalized to [0., 1.]
```
Comment (fchollet) [inline kerascv/layers/anchor_generators/anchor_generator.py:40]: In general, prefer fully-spelled out argument names, e.g. `normalize_coordinates`

File: kerascv/layers/anchor_generators/multi_scale_anchor_generator.py (hunk 2 above)
Comment (fchollet) [inline kerascv/layers/anchor_generators/multi_scale_anchor_generator.py:19]: Given the similarity with AnchorGenerator why not subclass it?

File: kerascv/layers/anchor_generators/anchor_generator.py [hunk 4]
```diff
@@ -1,7 +1,47 @@
...
+    # Attributes:
+        image_size: A list/tuple of 2 ints, the 1st represents the image height, the 2nd image width.
+        scales: A list/tuple of positive floats (usually less than 1.) as a fraction to shorter side of `image_size`.
+            It represents the base anchor size (when aspect ratio is 1.). For example, if `image_size` is (300, 200),
+            and `scales=[.1]`, then the base anchor size is 20.
+        aspect_ratios: a list/tuple of positive floats representing the ratio of anchor width to anchor height.
+            **Must** have the same length as `scales`. For example, if `image_size=(300, 200)`, `scales=[.1]`,
+            and `aspect_ratios=[.64]`, the base anchor size is 20, then anchor height is 25 and anchor width is 16.
+            The anchor aspect ratio is independent to the original aspect ratio of image size.
+        anchor_stride: A list/tuple of 2 ints or floats representing the distance between anchor points.
```
Comment (fchollet) [inline kerascv/layers/anchor_generators/anchor_generator.py:30]: Nit: why `anchor_stride` and `anchor_offset` rather than just `stride` and `offset`? It's an anchor generator, after all.

File: kerascv/layers/anchor_generators/anchor_generator.py (hunk 3 above)
Comment (tanzhenyu) [reply kerascv/layers/anchor_generators/anchor_generator.py]: Done.

File: kerascv/layers/anchor_generators/multi_scale_anchor_generator.py (hunk 2 above)
Comment (tanzhenyu) [reply kerascv/layers/anchor_generators/multi_scale_anchor_generator.py:19]: This is the class that accepts a list of things and then apply to each one of the anchor generator?

File: kerascv/layers/anchor_generators/anchor_generator.py (hunk 4 above)
Comment (tanzhenyu) [reply kerascv/layers/anchor_generators/anchor_generator.py]: Good point. Done.

File: kerascv/layers/anchor_generators/anchor_generator.py (hunk 1 above)
Comment (tanzhenyu) [reply kerascv/layers/anchor_generators/anchor_generator.py:18]: Yep and we intend to make them layers. Though the arguments are slightly different than this directory

**Comment by fchollet** [comment]
LGTM!
